xml/encoder_test.go

```diff
@@ -0,0 +1,424 @@
+package xml_test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/awslabs/smithy-go/xml"
+)
+
+func TestEncoder(t *testing.T) {
+	b := bytes.NewBuffer(nil)
+	encoder := xml.NewEncoder(b)
+
+	func() {
+		root := encoder.RootElement("Root", nil)
```

Comment:
If These took `StartElement` there wouldn't be a need for the `nil` second parameter. Generally attributes are the rare case in the protocol serialization.